### DISCUSSION TITLE
How do I create a new instance of a method object from a string?

### DISCUSSION THREAD

**Discussion by LiamM32** [julien-boudry/Condorcet#160] [Q&A] [answered]
I have a class derived from `Election` that I want to be able to have more control over a voting method object. But I'm struggling to understand how the `Election` object in Condorcet does this using a string.

I looked inside the `Election::getResult()` function, but I don't understand how it works. There appears to be an uninitialized variable [in this function](https://github.com/julien-boudry/Condorcet/blob/83cd3a94955f49405b0ffa4889a2e71d49f4cfa2/src/ElectionProcess/ResultsProcess.php#L80C34-L80C34) called `$wanted_method`, yet PHP never gives a warning. I would also be interested in hearing why this is.

But primarily, I would like to know how I can create an object for a voting method specified in a string. Is there a function for this?

**Comment by LiamM32** [comment]
I have found the answer to my primary question myself. To add the method to the election's `Calculator` array:
```
$method = Condorcet::getMethodClass($method);
$this->initResult($method);
````
Because `Election::initResult()` needs the path to the method. I suppose if you just wanted to make a method object, and have it stored wherever you want, this should work:
```
$method = Condorcet::getMethodClass($method);
$this->Calculator[$class] = new $class($this);
````
I'm not closing this discussion yet, as I'm still curious about that variable called `$wanted_method`, & I don't know where else to ask.

**Comment by julien-boudry** [comment, marked as answer]
Yes it should work.


The magic is just here, directly into the condition.
![image](https://github.com/julien-boudry/Condorcet/assets/4020317/8886030a-084c-435b-9ee4-af60c959bf47)

**Reply by LiamM32** [reply]
I noticed that. Though I was confused to see an assignment operator in a condition. Though I just found an article about what it means.

**Reply by LiamM32** [reply]
I don't like this that the language allows this, as I think it makes it too easy to assign a value by mistake.

Using a single equal sign as a condition is one my most common mistakes.

**Reply by julien-boudry** [reply]
The result of the assignment operation is evaluated in the condition. This means you don't need to perform an assignment if the previous conditions are met, and you save some code.
These are quite common in high-level languages.
